src/bitcoin-cli.cpp

```diff
@@ -466,9 +478,18 @@ class NetinfoRequestHandler : public BaseRequestHandler
         if (!args.empty()) {
             uint8_t n{0};
             if (ParseUInt8(args.at(0), &n)) {
-                m_details_level = std::min(n, MAX_DETAIL_LEVEL);
+                m_details_level = std::min(n, NETINFO_MAX_LEVEL);
             } else {
-                throw std::runtime_error(strprintf("invalid -netinfo argument: %s\nFor more information, run: bitcoin-cli -netinfo help", args.at(0)));
+                throw std::runtime_error(strprintf("invalid -netinfo level argument: %s\nFor more information, run: bitcoin-cli -netinfo help", args.at(0)));
+            }
+            if (args.size() > 1) {
+                if (std::string_view s{args.at(1)}; n && (s == "o" || s == "outonly")) {
+                    m_outbound_only_selected = true;
+                } else if (n) {
+                    throw std::runtime_error(strprintf("invalid -netinfo outonly argument: %s\nFor more information, run: bitcoin-cli -netinfo help", args.at(1)));
+                } else {
+                    throw std::runtime_error(strprintf("invalid -netinfo outonly argument: %s\nThe outonly argument is only valid for details levels higher than 0. For more information, run: bitcoin-cli -netinfo help", args.at(1)));
```

Comment:
nit: since `args.at(1)` is repeated also in the `throw` statements, maybe:

```cpp
std::string_view arg1{args.at(1)};
if (n && (arg1 == "o" || arg1 == "outonly")) {
} else {
    throw ... arg1 ...
```

Reply:
done